streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -23,11 +23,13 @@
 import java.util.Map;
 
 /**
- * The window specifications used for joins.
+ * The window specifications used for joins. TODO extend
  */
 public class JoinWindows extends Windows<TumblingWindow> {
 
+    /** Maximum time difference for tuples that are before the join tuple. */
```

Comment:
Yes, the time units in those windowing specs should not be defined by the framework, as they are based on event time.